cmd/certgen.go

```diff
@@ -228,6 +252,104 @@ func generateCertificates() error {
 		}
 	}
 
+	if option.Config.ExternalWorkloadCertsGenerate {
+		haveCAConfigMap := false
+		externalworkloadCA := generate.NewCA(option.Config.ExternalWorkloadCACertConfigMapName, option.Config.CiliumNamespace)
+
+		// Load CA from file?
+		if option.Config.ExternalWorkloadCACertFile != "" && option.Config.ExternalWorkloadCAKeyFile != "" {
+			log.Info("Loading ExternalWorkload CA from file")
+			err = externalworkloadCA.LoadFromFile(option.Config.ExternalWorkloadCACertFile, option.Config.ExternalWorkloadCAKeyFile)
+			if err != nil {
+				return fmt.Errorf("failed to load ExternalWorkload CA: %w", err)
+			}
+		} else {
+			// Does the configmap already exist?
+			ctx, cancel := context.WithTimeout(context.Background(), option.Config.K8sRequestTimeout)
+			defer cancel()
+			err = externalworkloadCA.LoadFromConfigMap(ctx, k8sClient)
+			if err != nil {
+				log.WithError(err).Info("failed to load ExternalWorkload configmap CA, generating new CA")
+				err = externalworkloadCA.Generate(option.Config.ExternalWorkloadCACertCommonName, option.Config.ExternalWorkloadCACertValidityDuration)
```

Comment:
nit: Should we check for `k8sErrors.IsNotFound` here before generating a new CA? If the service account has permission to create secrets, but no permissions to read secrets (which e.g. is currently the default in the Cilium Helm chart), we unconditionally overwrite an existing CA here.

Reply:
OK, good to know, will do.